packages/select-rich/test/lion-select-rich.test.js

```diff
@@ -0,0 +1,193 @@
+    });
+  });
+
+  describe('Invoker', () => {
+    it('should be shown closed by default', async () => {
+      const el = await fixture(html`
+        <lion-select-rich></lion-select-rich>
+      `);
+      expect(el._listboxElement.style.display).to.be.equal('none');
+    });
+
+    it('can toggle the listbox', async () => {
+      const el = await fixture(html`
+        <lion-select-rich></lion-select-rich>
+      `);
+      expect(el._listboxElement.style.display).to.be.equal('none');
+      el.click();
+      expect(el._listboxElement.style.display).to.be.equal('inline-block');
+    });
+
+    describe('Keyboard navigation', () => {
+      it('opens the listbox with [enter] key', async () => {
+        const el = await fixture(html`
+          <lion-select-rich></lion-select-rich>
+        `);
+        el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
+        await el.updateComplete;
+        expect(el._listboxElement.style.display).to.be.equal('inline-block');
+      });
+
+      it('opens the listbox with [space] key', async () => {
+        const el = await fixture(html`
+          <lion-select-rich></lion-select-rich>
+        `);
+        el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Space' }));
+        await el.updateComplete;
+        expect(el._listboxElement.style.display).to.be.equal('inline-block');
+      });
+
+      it('opens the listbox with [arrow down] key and changes focus to listbox', async () => {
+        const el = await fixture(html`
+          <lion-select-rich></lion-select-rich>
+        `);
+        el.dispatchEvent(new KeyboardEvent('keydown', { key: 'ArrowDown' }));
+        await el.updateComplete;
+        expect(el.querySelector('[slot="content"]').style.display).to.be.equal('inline-block');
+        expect(el._listboxElement.focus).to.be.true;
+      });
+
+      it('closes the listbox with [enter] key once opened', async () => {
+        const el = await fixture(html`
+          <lion-select-rich></lion-select-rich>
+        `);
+        el.click();
+        expect(el._listboxElement.style.display).to.be.equal('inline-block');
+        el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
+        await el.updateComplete;
+        expect(el._listboxElement.style.display).to.be.equal('none');
+        expect(el._invokerElement.focus).to.be.true;
+      });
+
+      it('closes the listbox with [esc] key once opened', async () => {
+        const el = await fixture(html`
+          <lion-select-rich></lion-select-rich>
+        `);
+        el.click();
+        expect(el._listboxElement.style.display).to.be.equal('inline-block');
+        el.dispatchEvent(new KeyboardEvent('keydown', { key: 'Esc' }));
+        await el.updateComplete;
+        expect(el._listboxElement.style.display).to.be.equal('none');
+        expect(el._invokerElement.focus).to.be.true;
+      });
+
+      it('closes the listbox with [tab] key once opened, focuses on next element', async () => {
```

Comment:
May also want to add a test for [tab] that the listbox will have the last focused option as the selected one as it loses focus and closes. This is what happens at least in the WAI-ARIA

Reply:
That is sort of already tested here: https://github.com/ing-bank/lion/pull/74/files#diff-3a1ca2a47600fe15d53986d5c9665db3R58, but adding a test like `selects an option on setting focus` there makes sense to me.

Reply:
Ah yea that should cover it